### PR TITLE
Fix spikes breaking Multiscale

### LINKED ISSUE #188: Spikes break Dragonite's Multiscale
When you put up 1 layer of spikes, it says that Dragonite's Multiscale is broken on the calc.

### PR DESCRIPTION
**Pull request by Ntonio36**
Closes #188

### CODE REVIEW

**Comment by AustinXII** [comment]
You should be checking for the Pokémon’s typing, mulitscale isn’t immune to spikes unless they are part flying type

**Comment by Ntonio36** [comment]
Aaand done.

File: js/damage.js [hunk 1]
```diff
@@ -629,7 +629,7 @@ function getDamageResult(attacker, defender, move, field) {
 		finalMods.push(field.format !== "Singles" ? (gen >= 6 ? 0xAAC : 0xA8F) : 0x800);
 		description.isLightScreen = true;
 	}
-	if ((defAbility === "Multiscale" || defAbility === "Shadow Shield") && defender.curHP === defender.maxHP && !field.isSR && (!field.spikes || (field.spikes && defAbility === "Multiscale"))) {
+	if ((defAbility === "Multiscale" || defAbility === "Shadow Shield") && defender.curHP === defender.maxHP && !field.isSR && (!field.spikes || (defAbility === "Multiscale" && field.spikes && [defender.type1, defender.type2].indexOf("Flying") != -1))) {
```
Comment (Zarel) [inline js/damage.js]: You don't `defAbility === "Multiscale" &&`

File: js/damage.js [hunk 2]
```diff
@@ -629,7 +629,7 @@ function getDamageResult(attacker, defender, move, field) {
 		finalMods.push(field.format !== "Singles" ? (gen >= 6 ? 0xAAC : 0xA8F) : 0x800);
 		description.isLightScreen = true;
 	}
-	if ((defAbility === "Multiscale" || defAbility === "Shadow Shield") && defender.curHP === defender.maxHP && !field.isSR && !field.spikes) {
+	if ((defAbility === "Multiscale" || defAbility === "Shadow Shield") && defender.curHP === defender.maxHP && !field.isSR && (!field.spikes || (defAbility === "Multiscale" && field.spikes && [defender.type1, defender.type2].indexOf("Flying") != -1))) {
```
Comment (Zarel) [inline js/damage.js]: You don't need `defAbility === "Multiscale" &&`

Comment (Ntonio36) [reply js/damage.js]: Ah yes, my bad. Fixing it right away.

Comment (Zarel) [reply js/damage.js]: The entirety of

`(!field.spikes || (defAbility === "Multiscale" && field.spikes && [defender.type1, defender.type2].indexOf("Flying") != -1))`

is overcomplicated, really.

`(!field.spikes || [defender.type1, defender.type2].indexOf("Flying") != -1)`

does the exact same thing, although you can optimize a bit with

`(!field.spikes || defender.type1 === "Flying" || defender.type2 === "Flying")`

**Comment by Zarel** [comment]
Looks good now. @AustinXII?

**Comment by AustinXII** [comment]
Looks fine, I just wanted to make sure it worked how expected when testing.

**Pull request merged**